### ISSUE TITLE
Does KSV WebRTC Support The Analyzation/Processing of Real Time AWS Chime Video Data? 

### ISSUE THREAD

**Issue by mliu0515** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1223]
**Design considerations** 
I am trying to build a video chat web application, and I am thinking about using AWS Chime sdk for building the application. On top of the video chat, I also want to apply ML model onto the Chime video data in a real time setting. I understand that KVS WebRTC gives me the capability to live stream media or perform two-way audio or video interaction along with Real time media streaming, thus I want to know if KVS can perform real time processing and analysis of real time video streaming data generated by a Chime backed meeting session? Much advice would be appreciated!

**TLDR**
I am looking for a possible solution to process/analyze real time AWS Chime video data, and I wonder whether or not KVS can provide me with a solution.

**Comment by hassanctech** [comment]
No, KVS does not do analysis or processing on the data. 

Also KVS provides 2 classes of SDKs webrtc and producer:

1. WebRTC, this is peer to peer, this video does't get stored anywhere it's just real time video and we do not offer any processing integration, your application would need to do whatever processing it needs on the video
2. Producer, this video gets stored in our back-end and then you can request it using our GetMedia APIs, this is NOT a peer to peer offering.  By default we ingest the video and vend it back to you exactly the way that you put it in.  There are integrations with services like AWS Rekognition, here are some resources to explore:

https://docs.aws.amazon.com/rekognition/latest/dg/streaming-video.html
